warehouse/query-core/src/main/java/datawave/query/transformer/FacetedTransformer.java

```diff
@@ -29,31 +34,37 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.log4j.Logger;
 
-public class FacetedTransformer extends DocumentTransformer {
+public class FacetedTransformer extends BaseQueryLogicTransformer<Entry<Key,Value>,FacetsBase> {
     
     private static final Logger log = Logger.getLogger(FacetedTransformer.class);
     
+    private DocumentTransformer documentTransformer;
+    
```

Comment:
FacetedTransformer has the most disruptive changes. It no longer extends DocumentTransformer but instead creates and uses a DocumentTransformer as a delegate, then does its own transform with appropriate input and output types.